widget/select.go

```diff
@@ -295,36 +295,39 @@ func (s *Select) updateSelected(text string) {
 }
 
 type selectRenderer struct {
-	*widget.ShadowingRenderer
+	icon             *Icon
+	label            *textProvider
+	background, line *canvas.Rectangle
 
-	icon       *Icon
-	label      *textProvider
-	background *canvas.Rectangle
-	combo      *Select
+	objects []fyne.CanvasObject
+	combo   *Select
 }
 
+func (s *selectRenderer) Objects() []fyne.CanvasObject {
+	return s.objects
+}
+
+func (s *selectRenderer) Destroy() {}
+
 // Layout the components of the button widget
 func (s *selectRenderer) Layout(size fyne.Size) {
-	doublePad := theme.Padding() * 2
-	s.LayoutShadow(size.Subtract(fyne.NewSize(doublePad, doublePad)), fyne.NewPos(theme.Padding(), theme.Padding()))
-	inner := size.Subtract(fyne.NewSize(doublePad*2, doublePad))
+	s.line.Resize(fyne.NewSize(size.Width, theme.InputBorderSize()))
+	s.line.Move(fyne.NewPos(0, size.Height-theme.InputBorderSize()))
+	s.background.Resize(fyne.NewSize(size.Width, size.Height-theme.InputBorderSize()*2))
+	s.background.Move(fyne.NewPos(0, theme.InputBorderSize()))
 
-	s.background.Move(fyne.NewPos(theme.Padding(), theme.Padding()))
-	s.background.Resize(size.Subtract(fyne.NewSize(doublePad, doublePad)))
-
-	offset := fyne.NewSize(theme.IconInlineSize(), 0)
-	labelSize := inner.Subtract(offset)
+	iconPos := fyne.NewPos(size.Width-theme.IconInlineSize()-theme.Padding()*2, (size.Height-theme.IconInlineSize())/2)
+	labelSize := fyne.NewSize(iconPos.X-theme.Padding(), s.label.MinSize().Height)
 
+	// TODO are these needed?
```

Comment:
Is there something additional to do here?

Reply:
Yes, I would like to know why is needed to use mutex lock in this part (line 323):

```
         s.combo.propertyLock.RLock()
	defer s.combo.propertyLock.RUnlock()

	s.label.Resize(labelSize)
	s.label.Move(fyne.NewPos(doublePad, theme.Padding()))
	s.label.Move(fyne.NewPos(theme.Padding(), (size.Height-labelSize.Height)/2))
```

Reply:
We went through a process of hunting race conditions a while back. I cannot immediately see what this was protecting, perhaps “test -race” will show something extra if they are removed? If not then probably it is historic.

Reply:
I have just run `go test -race ./...` and there are a lot of race conditions, is that expected?

Reply:
Btw, I have run now `go test -race ./... | grep Select` to filter just Select related tests (after removing the mutex lock) and everything seems to be ok, so I think it is save to remove that piece of code.

Reply:
> I have just run `go test -race ./...` and there are a lot of race conditions, is that expected?

yes, the widget renderer access had to be disabled (in 2.0) before I can fix another class of race condition (there is a ticket to track these #1028)

Reply:
Oh ok great! :)